crates/ra_editor/src/lib.rs

```diff
@@ -124,11 +126,27 @@ fn check_unnecessary_braces_in_use_statement(file: &SourceFileNode) -> Vec<Diagn
     let mut diagnostics = Vec::new();
     for node in file.syntax().descendants() {
         if let Some(use_tree_list) = ast::UseTreeList::cast(node) {
-            if use_tree_list.use_trees().count() <= 1 {
+            if use_tree_list.use_trees().count() == 1 {
+                let range = use_tree_list.syntax().range();
+                // use_tree_list always has one child, so we use unwrap directly here.
+                let to_replace = typing::single_use_tree(use_tree_list)
+                    .unwrap()
+                    .syntax()
+                    .text()
+                    .to_string();
+                let mut edit_builder = TextEditBuilder::new();
+                edit_builder.delete(range);
+                edit_builder.insert(range.start(), to_replace);
```

Comment:
This doesn't seem to handle `self` nicely. It's very rare that this would come up, but it could. See #316.

Reply:
`use a::b::{self}` should be fixed as `use a::b` instead of `use a::b::self`?

Reply:
Yeah, this can be handled in a separate PR